proposals/XX-bw-xxx.txt

```diff
@@ -0,0 +1,202 @@
+  contributing on pastly's new bandwidth scanner implementation.
+
+  This specification was revised after feedback from:
+
+    XXX
+
+1.3 Outline
+
+  The bandwidth measurements mentioned in sections 3.4.1 and 3.4.2
+  of "Tor directory protocol" (dir-spec.txt) [3] are obtained
+  by bandwidth authorities, which generate a file storing information
+  on relays' measured bandwidth capacities.
+
+2. Format details
+
+  Bandwidth measurements MUST contain the following sections:
+  - Header (exactly once)
+  - Relays measurements (zero or more times)
+
+2.1. Definitions
+
+  The following nonterminals are defined in dir-spec.txt, sections
+  1.2., 2.1.1., 2.1.3. :
+
+    Int
+    SP (space)
+    NL (newline)
+    Keyword
+    ArgumentChar
+    fingerprint (hexdigest)
+    nickname
+
+  Nonterminals defined in "Tor Directory List Format" (dir-list-spec.txt),
+  section 2.2.1.:
+
+    version_number
+
+  We define the following nonterminals:
+
+    value ::= ArgumentChar+
+    key_value ::= Keyword "=" value
+    line ::= ArgumentChar* NL
+    timestamp ::= Int
+    bandwidth ::= Int
+    relay_line ::= key_value (SP key_value)* NL
+
+2.2. Header format
+
+It consists of:
+
+  timestamp NL
+
+    [At start, exactly once.]
+
+    The Unix Epoch time in seconds when the file was created.
+
+  "version="version_number NL
+
+    [In second position, zero or one time.]
+
+    The specification document format version.
+    It uses semantic versioning [5].
+    This line has been added in version 1.1.0 of this specification.
+    Documents generated before this specification do not contain this line,
+    and the version_number is considered to be "1.0.0".
+
+  "software=" value NL
+
+    [Zero or one time.]
+
+    The name of the software that created the document.
+
+    This line has been added in version 1.1.0 of this specification.
+    Documents generated before this specification do not contain this line,
+    and the software is considered to be "torflow".
+
+  "software_version=" version_number NL
+
+    [Zero or one time.]
+
+    The version of the software that created the document.
+    It uses semantic versioning.
+    This line has been added in version 1.1.0 of this specification.
+
+  "scanner_started=" timestamp NL
+
+    [Zero or one time.]
+
+    The Unix Epoch time in seconds when the file was created.
+
+  "earliest_measurement=" timestamp NL
+
+    [Zero or one time.]
+
+    The Unix Epoch time in seconds when the file was created.
+
+  key_value NL
+
+    [Zero or more times.]
+
+    Future releases may include additional key_value header lines.
+    Parsers MUST NOT rely on the order of these additional lines.
+    Additional header lines will be accompanied by a minor version increment.
+    Additional header lines MUST NOT use any keywords specified in the
+    relay measurements format.
+
+  NL
+
+    [One time.]
+
+    The header ends.
+    For documents generated before this specification, the header ends
```

Comment:
Please say "version 1.0.0 documents", because torflow will continue to generate documents without new lines after this specification is merged.